content/v3/git/refs.md

```diff
@@ -14,11 +14,26 @@ The `ref` in the URL must be formatted as `heads/branch`, not just `branch`. For
 
     GET /repos/:owner/:repo/git/refs/heads/skunkworkz/featureA
 
-### Response
-
 <%= headers 200 %>
 <%= json :ref %>
 
+If the ref doesn't exist in the repository, but existing refs start with ref
+they will be returned as an array. For example, a call to get the data for a
+branch named `feature`, which doesn't exist, would return head refs
+including `featureA` and `featureB` which do.
+
+    GET /repos/:owner/:repo/git/refs/heads/feature
+
+<%= headers 200 %>
+<%= json :refs_matching %>
+
+If the ref doesn't match an existing ref or any prefixes a 404 will be returned.
+
+    GET /repos/:owner/:repo/git/refs/heads/ref-that-like-for-sure-totally-doesnt-exist
```

Comment:
Would you mind changing this example branch name? Let's just go with something that still gets the point across but in a more... universal voice :smile: Perhaps: "`GET /repos/:owner/:repo/git/refs/heads/feature-branch-that-no-longer-exists`